### DISCUSSION TITLE
Password expiration

### DISCUSSION THREAD

**Discussion by sp132** [zitadel/zitadel#6660] [Q&A] [answered]
Hi!

Please help me to understand something. I'm using a self-hosted instance of a Zitadel server with a custom UI and need to implement password expiration check (and subsequent password change prompt).

I've found an [API](https://zitadel.com/docs/apis/resources/mgmt/management-service-update-custom-password-age-policy) that allows me to set max password age, but I'm not sure how this policy is enforced (if at all) or what would be the indication of an expired password. Documentation is suspiciously lacking any explanation on this, so maybe password expiration is considered a "legacy" feature that is no longer supported?

**Comment by livio-a** [comment]
Hey @sp132
Sorry  i think you really hit a "legacy" feature /  left-over there, which is not properly documented
<img width="470" alt="image" src="https://github.com/zitadel/zitadel/assets/9405495/9c671d87-ca51-43a2-a4a0-75202f55fc47">

As presumed by you, it's currently not enforced nor is there an indication of password being expired.

To reduce the confusion, we'll remove it from the api.

**Reply by livio-a** [reply, marked as answer]
just noticed that it's also related to https://github.com/zitadel/zitadel/issues/6614, resp. with the implementation of https://github.com/zitadel/zitadel/issues/6434 an implementation might then at least be able to check when the password was changed the last time. The password age policy could then be used (on client / login side) to determine if the user needs to change the password, while not being enforce by zitadel itself.

**Reply by sp132** [reply]
@livio-a got you, thanks!
